### ISSUE TITLE
Make thread names of CoroutineScheduler more distinguishable

### ISSUE THREAD

**Issue by qwwdfsad** [Kotlin/kotlinx.coroutines#4267]
Currently, names of all threads that belong to `CoroutineScheduler` (`Dispatchers.Default`, `Dispatchers.IO` and `Dispatchers.IO.limitedParallelism(...)` are named `DefaultDispatcher-worker-*` which might be quite confusing to look at in thread dumps (i.e. it's possible to have dozens of such threads on 8 core machine).
Example of such confusion: https://stackoverflow.com/questions/78502056/why-thread-name-is-defaultdispatcher-worker-even-though-i-specify-function-to-ru

While we cannot leverage `Thread.setName` because of its performance impact, we still can sweeten the pill:
* Name the thread somehow more generic (for example: `CoroutineScheduler-worker-`, something like "shared-pool-worker-` etc.)
* Add an artificial stackframe that indicates that the thread was spawned and/or is parked as part of the "IO" pool if it's possible implementation-wise

**Comment by ck2e14** [comment]
Agreed. As a relatively new Kotlin dev this definitely tripped me up. I should add that the Kotlin docs are really clear about this thread sharing behaviour, so thanks for that! That said... for semantics' sake:

Is it an overreach of naming if IO dispatcher and Default dispatcher share the pool? Why not call the threads 'shared-pool-worker-x' as [qwwdfsad](https://github.com/qwwdfsad) suggests, and then we can just name our scopes such that we know what the dispatcher was with a prefix or something?
